Kuadrant/kuadrant-controller
# Handle httproute signal from ratelimitpolicy controller

This PR:
- Removes `ObjectRefStatus` and uses `<namespace/name>` string format to store gateways for consistency.
- Generalizes RLP's status block to include `HTTPRoute`s as well now.
- Moves signaling event from predicate to (httproute/virtualservice) controllers.
## Verification Steps
1. `make local-setup`
2. `kubectl apply -f examples/toystore/toystore.yaml`
3. `kubectl apply -f examples/toystore/httproute.yaml`
4. `kubectl apply -f examples/toystore/ratelimitpolicy.yaml`
5. `kubectl annotate httproute/toystore kuadrant.io/ratelimitpolicy=toystore`
6. `k port-forward -n kuadrant-system svc/kuadrant-gateway 9080:80`
7. `curl -v -H 'Host: api.toystore.com' http://localhost:9080/toy` 1 time and expect `429` response.
 
Document change: code-level
